src/foam/nanos/ticket/Ticket.js

```diff
@@ -257,6 +262,61 @@ foam.CLASS({
     }
   ],
 
+  methods: [
+    {
+      name: 'authorizeOnCreate',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaThrows: ['AuthorizationException'],
+      javaCode: `
+        // Everyone can create a ticket
+      `
+    },
+    {
+      name: 'authorizeOnRead',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaThrows: ['AuthorizationException'],
+      javaCode: `
+        Subject subject = (Subject) x.get("subject");
+        User user = subject.getRealUser();
+
+        if ( user.getId() != this.getCreatedBy() ) {
+          throw new AuthorizationException("You don't have permission to read this ticket.");
```

Comment:
why do we not check for ticket.read.id permission at this point?  like you do for update?

Reply:
We have "ticket.read.*" for msp-admin and it is already taken care of by checkGlobalRead. Do you think we need to also support for giving any other user permission to access a particular ticket.id?

Reply:
I'm more about being inconsistent. 
we do an auth check on update.id but not for read.  
Please add a an auth check for ticket.read.id.

Reply:
OK